### PR TITLE
Phoenix: adds a new macro, that validates that the arguments which it takes are 0..10_000

### PR DESCRIPTION
**Pull request by gangov**

### CODE REVIEW

File: packages/phoenix/src/utils.rs [hunk 1]
```diff
@@ -122,4 +135,21 @@ mod tests {
         let tolerance = Decimal::percent(3);
         assert!(!assert_approx_ratio(a, b, tolerance));
     }
+
+    #[test]
+    #[should_panic(expected = "The value -1 is out of range. Must be between 0‱ and 10000‱")]
+    fn test_below_min() {
```
Comment (ueco-jb) [inline packages/phoenix/src/utils.rs:141]:
```suggestion
    fn validate_bps_below_min() {
```
and so on.

**Review by ueco-jb — COMMENTED** [review]
Implement usage of this macro in a factory for example.

**Comment by gangov** [comment]
should the new macro be used solely in factory, or we can use that in the other places in our contracts where we deal with bps?

**Comment by ueco-jb** [comment]
> should the new macro be used solely in factory, or we can use that in the other places in our contracts where we deal with bps?

Add it in `packages/phoenix/utils.rs`.

**Comment by gangov** [comment]
it's already there, I meant to ask if we should use this macro and verify all the places where we have bps in our DEX with it

**Comment by ueco-jb** [comment]
> if we should use this macro and verify all the places where we have bps in our DEX with it

In all the places where a new parameter in a BPS format is passed. Factory instantiation has its own validation and it shares some of those parameters with pools and staking contracts.
Then you don't need to validate the same parameter again in the factory, but if some pool parameters are passed during each instantiation then they should be validated.

**Review by ueco-jb — APPROVED** [review]
LGTM

**Pull request merged**